ERCS/erc-7753.md

```diff
@@ -0,0 +1,88 @@
+---
+eip: 7753
+title: Algorithmically generated NFT
+description: The image of the NFT is represented through mathematical algorithms.
+author: LEVI_104 (@chen4903), birch (@birchwork)
+discussions-to: https://ethereum-magicians.org/t/algorithmically-generated-nft/20761
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-08-08
+requires: 165, 721
+---
+
+## Abstract
+
+A non-fungible token standard that uses mathematical algorithms to draw and store images without any servers or storage space.
+
+## Motivation
+
+The current NFT solutions store images on IPFS or third-party servers, consuming storage space and being separate from the blockchain. Our goal is to represent image information using mathematical algorithms stored in smart contracts, tightly integrating with the blockchain. This approach uses no storage space and allows anyone to render the image information at any time based on the algorithm stored in the smart contract.
+
+## Specification
+
+### Interface
+
+This standard requires the implementation of the following interface:
+```solidity
+pragma solidity ^0.8.0;
+
+/// @title ERC-7790 Algorithmically Generated Token Standard
+interface IERC7790 {
+    /// @dev When a new token is created, emit this event to record the algorithm content.
+    event SetTokenAlgorithm(uint256 indexed tokenID, string algorithm);
+
+    /// @dev The project team or administrator manually inputs the mathematical 
+    /// algorithm for the corresponding token, with the parameter algorithm adhering 
+    /// to the format requirements specified by this EIP.
+    function setTokenAlgorithm(uint256 tokenID, string memory algorithm) external returns(bool);  
+    
+    /// @dev Returns the mathematical algorithm for the token image based on
+    /// the token ID. The returned data is a JSON text that adheres to the format specified by this EIP.
+    function getTokenAlgorithm(uint256 tokenID) external view returns(string memory);
+}
+```
+### Format Requirements
+
+```json
+{
+    "Name": "Asset name",
+    "Encoding Method": "Specify the string encoding and decoding method for storing the algorithm",
+    "properties": [
+       "Mathematical Formula 1",
+       "Mathematical Formula 2",
+       "...", 
+    ]
+}
+```
+
+## Rationale
+
+This EIP aims to address the data storage issues of ERC-721. Currently, [ERC-721](https://eips.ethereum.org/EIPS/eip-721) uses `tokenURI()` to return a URL pointing to the image's storage location, allowing users to access the image via this URL. However, this method consumes storage space and relies on network access. We propose a solution: using mathematical algorithms to depict image information. Anyone can reproduce the image based on the algorithm, without relying on any off-chain storage, and can recreate the image using any mathematical tool.
```

Comment:
This content is describing the proposal as a whole (and so belongs in the Motivation section.) The Rationale section exists to explain choices made within the proposal. My favourite analogy is:

> Motivation: We decided to build a shed because...
> Rationale: We decided to paint the shed red because...